### PR TITLE
Update version and CHANGELOG for v1.12.0

### PR DESCRIPTION
**Pull request by EmilyBourne**
Update the version in `version.py` to 1.12.0. Update the CHANGELOG to move the changes in #1858 to a new section for version 1.12.0.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -72,18 +62,42 @@ All notable changes to this project will be documented in this file.
...
 -   \[INTERNALS\] Remove unused parameters `expr`, `status` and `like` from `pyccel.ast.core.Assign`.
 -   \[INTERNALS\] Remove `pyccel.ast.utilities.builtin_functions`.
 -   \[INTERNALS\] Remove unused/unnecessary functions in `pyccel.parser.utilities` : `read_file`, `header_statement`, `accelerator_statement`, `get_module_name`, `view_tree`.
 -   \[INTERNALS\] Remove unused functions `Errors.unset_target`, and `Errors.reset_target`.
+
+## \[1.12.0\] - 2024-05-08
+
+### Added
+
+-   #1830 : Add a `pyccel.lambdify.lambdify` function to accelerate SymPy expressions.
```
Comment (yguclu) [inline CHANGELOG.md:80]: Is this not accessible directly as `pyccel.lambdify`?

Comment (EmilyBourne) [reply CHANGELOG.md:80]: Yes, and it is not accessible as `pyccel.lambdify.lambdify`. This should have been caught in the PR. I will fix it here

Comment (EmilyBourne) [reply CHANGELOG.md]: [d752b18](https://github.com/pyccel/pyccel/pull/1859/commits/d752b181ca240273a3d423acf19cb9d98690bd12)

**Pull request merged**